### DISCUSSION TITLE
How to get to show Loras/HNs in the autocomplete?

### DISCUSSION THREAD

**Discussion by Panchovix** [DominikDoom/a1111-sd-webui-tagcomplete#117] [Q&A] [answered]
Hi there guys, really nice extension! 

Saw that recently, it got support for Hypernetworks and LoRAs (I guess because after the addition of <lora:Lora ame: weight> recently on the webui)

I was wondering, how I would be able to the autocomplete to show them? "<" by default shows only embeddings, and tried "<h:" and "<l:" but no luck either.

Thanks in advance!

**Comment by DominikDoom** [comment, marked as answer]
If you have updated the extension, it should already be visible. This is what it looks like for me when typing `<`:
![image](https://user-images.githubusercontent.com/34448969/214596654-39851082-7f7b-4287-af75-56283391f8fa.png)

The only prerequisites should be (obviously) that the hypernetworks and LoRAs are in the default folders or whatever you specified as your hypernetwork and lora directories. 

If the webui shows them in the extra networks section ![image](https://user-images.githubusercontent.com/34448969/214598339-90ed24dd-7b04-4f07-a357-0f226be7fa60.png), the script should be able to find them too. You could also check in `extensions\<tag-autocomplete>\tags\temp` if hyp.txt and lora.txt exist and were filled correctly.

If you are on the latest extension version and checked the above, it sounds like a bug. Any errors in the browser console or terminal maybe?

**Reply by Panchovix** [reply]
When using <, I only get this
![image](https://user-images.githubusercontent.com/19153797/214601852-744c5333-3f3b-49e2-a928-f1cd07af8ceb.png)

The webui on the 
![image](https://user-images.githubusercontent.com/19153797/214601930-2bd5ef14-5471-42e9-9c8a-ffb5c7827990.png) button works, as it shows all embeds, Hns and LoRAs (Here an example of a character that I have embeds and LoRAs, the autocomplete shows the embedding only)
![image](https://user-images.githubusercontent.com/19153797/214602573-f1d26d55-6610-4f0a-8dc2-fc2a0b581528.png)

On `extensions\<tag-autocomplete>\tags\temp`, hyp.txt and lora.txt were filled correctly.
![image](https://user-images.githubusercontent.com/19153797/214602979-62030cee-977a-46aa-96c5-cef6afc5cb70.png)

I'm on latest commit of both extension and webui.
![image](https://user-images.githubusercontent.com/19153797/214603944-51ddc73f-21a4-4c25-ae03-5cf055e2aabe.png)

No errors on the terminal, haven't noticed an issue on the browser console either.

**Reply by DominikDoom** [reply]
Very strange, if the temp files were filled the browser side script should display them too.
Can you try just typing "loras" in your browser console? That should show the array the script uses if it loaded at all:
![image](https://user-images.githubusercontent.com/34448969/214604879-30da367b-a712-45b9-8746-e3a285247286.png)

Force-reloading the page with CTRL+F5 might also be worth a try to rule out cache issues.

**Reply by Panchovix** [reply]
> Very strange, if the temp files were filled the browser side script should display them too. Can you try just typing "loras" in your browser console? That should show the array the script uses if it loaded at all: ![image](https://user-images.githubusercontent.com/34448969/214604879-30da367b-a712-45b9-8746-e3a285247286.png)
> 
> Force-reloading the page with CTRL+F5 might also be worth a try to rule out cache issues.

Tried using "loras" in console and it gave me an error.
But, after forcing new cache with CTRL + F5, it worked on Chrome.

When accesing from a phone, had to remove the cache of the browser app, and it worked as well.

![image](https://user-images.githubusercontent.com/19153797/214608727-b03201a0-118e-4b15-bede-3169338ea38d.png)


Really thanks for your help!

**Reply by DominikDoom** [reply]
Alright, glad that was it! Seems like the browser cached an old version of the script then.
Sadly I don't really know of a way to prevent this for good, I already force the temporary .txt files to reload each time, but the script itself is out of my control since it is added to the page by the webui extension system.
